python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -367,10 +436,19 @@ def find_never_materialized_or_requested_root_asset_partitions(
     newly_materialized_root_asset_keys = set()
     newly_materialized_root_partitions_by_asset_key = defaultdict(set)
 
-    for asset_key in target_asset_keys & asset_graph.root_asset_keys:
+    for asset_key in (
+        AssetSelection.keys(*target_asset_keys) & AssetSelection.all().sources()
```

Comment:
`sources` is deprecated in favor of `roots`

Reply:
Also - why make this change - seems like a more verbose way of doing the same thing?